test/box/gh-6405-discarding-arguments-in-box-func-call.test.lua

```diff
@@ -0,0 +1,45 @@
+args_type = function(...)                     \
+    local type_table = {}                     \
+    for i = 1, select('#', ...) do            \
+        type_table[i] = type(select(i, ...))  \
+    end                                       \
+    return type_table                         \
+end                                           \
+
+function TablesAreEqual(table1, table2)       \
+    for k, v in pairs(table1) do              \
+        if table2[k] ~= v then                \
+            return false                      \
+        end                                   \
+    end                                       \
+                                              \
+    for k, v in pairs(table2) do              \
+        if table1[k] ~= v then                \
+            return false                      \
+        end                                   \
+    end                                       \
+                                              \
+    return true                               \
+end                                           \
+
+box.schema.func.create('args_type')
+
+box_call = box.func.args_type:call({1, nil, 3})
+lua_call = args_type(1, nil, 3)
+assert (TablesAreEqual(box_call, lua_call))
```

Comment:
Please move this logic to a separate function instead of copy-paste.

Reply:
I agree that a separate function would be nice, but then it would take a variable number of arguments and would be quite similar to box.func:call(), the argument passing to which I am testing. I think it's better to avoid repeating in the tests the case I'm testing with these tests.

Reply:
It is no similar to `box.func:call()`:
```
function test_process(...)
    local args_table = {}   
    for i = 1, select('#', ...) do   
        args_table[i] = select(i, ...)
    end 
    box_call = box.func.args_type:call(args_table)
    lua_call = args_type(1, nil, 3)
    assert (table.equals(box_call, lua_call))
end
```